groups/bsl/bslstl/bslstl_variant.h

```diff
@@ -11,13 +11,131 @@ BSLS_IDENT("$Id: $")
+// the storage of the 'bsl::variant' object.
+//
+// A program that instantiates the definition of variant with no template
+// arguments is ill-formed.
+//
+// A 'bsl::variant<TYPES...>' object can be created using copy/move
+// construction, and the resulting object holds the same alternative.  A
+// 'bsl::variant<TYPES...>' can also be created from a value of one of the
+// alternatives, or from a value of type for which there is an unambiguous best
+// match conversion to one of the alternatives.  In addition, 'bsl::variant'
+// also provides construction of an explicitly specified alternative from a
+// variadic number of arguments.
+//
+// If at least one alternative is allocator aware, 'bsl::variant' is allocator
+// aware.  For an allocator aware 'bsl::variant', each constructor has a
+// matching allocator extended version which specifies the allocator used for
+// the duration of the 'bsl::variant' object and which will be provided at
+// construction of any alternative created during the lifetime of the
+// 'bsl::variant<TYPES...>' object.  Note that the 'bsl::variant' object itself
+// does not allocate any memory, and that the alternative objects are created
+// *in*-*place*.
+//
+// A 'bsl::variant<TYPES...>' object can be copy/move assigned from another
+// 'bsl::variant<TYPES..>' object.  The destination 'bsl::variant' object will
+// have the same alternative as the source object.  Additionally, a
+// 'bsl::variant' object can be copy move assigned from a value of one of the
+// alternatives, or from a value of type for which there is an unambiguous best
+// match conversion to one of the alternatives.
+//
+// Lastly, 'bsl::variant<TYPES...>' provides 'emplace' methods, where the
+// alternative type is explicitly specified.
+//
+// Any alternative construction or assignment that results in an exception
+// being thrown will result in a 'bsl::variant' object that holds no value,
+// referred to as valueless by exception state, indicated by
+// 'valueless_by_exception()' method returning 'true' and 'index()' method
+// returning 'bsl::variant_npos' value.
+//
+// Two 'bsl::variant<TYPES...>' compare equal if they hold the same alternative
+// type and their contained value is the same, or they both hold no value.
+//
+// Currently active alternative can be checked using the 'index()' method which
+// returns the index of the current alternative.  Additionally, free function
+// 'holds_alternative' can be used to check whether an explicitly specified
+// type is the currently active alternative, as specified by the C++ standard.
+//
+// Retrieving the contained value is done using free functions 'get' and
+// 'get_if' as specified by the C++ standard.
+//
+// Free function 'visit' is provided that implements the visitor design pattern
+// as specified by the C++ standard, modulo limitations listed below.
+//
 //
 ///Usage
-///
+///-----
+// The following snippets of code illustrate use of this component:
```

Comment:
```suggestion
// This section illustrates intended use of this component.
```
https://bde.bloomberg.com/bde/knowledge_base/coding_standards.html#rule-8-3-2